src/libraries/StableMath.sol

```diff
@@ -51,6 +51,36 @@ library StableMath {
     }
     }
 
+    function _getAmountIn(
```

Comment:
Where did this impl come from?

Reply:
from uniswap / vexchange v2's library. Do we need to attribute it or something?

Reply:
This is a stable math impl so cant be uni-v2, was more concerned with its correctness?

Reply:
so this is adapted from:

1. _getAmountOut in the same library
2. Referenced Balancer's implementation [here](https://github.com/balancer-labs/balancer-v2-monorepo/blob/1f07236cd2d3a01fe687c1703b989c6284be2202/pkg/pool-stable/contracts/StableMath.sol#L164) 

But definitely I need to write tests to ensure its correctness

Reply:
implementation corrected with tests written, can review